File: 104. Maximum Depth of Binary Tree/solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def maxDepth(self, root):
        # If you get to the end of a branch, add nothing.
        if root is None:
            return 0
        
        # Call left, call right
        left = self.maxDepth(root.left)
        right = self.maxDepth(root.right)
        
        # Whichever returns bigger, return that plus one.
        return left + 1 if left > right else right + 1

File: 104. Maximum Depth of Binary Tree/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func maxDepth(root *TreeNode) int {
    if root == nil {
        return 0
    } else {
        left := maxDepth(root.Left)
        right := maxDepth(root.Right)
        if left > right {
            return left + 1
        } else {
            return right + 1
        }
    }
}
